### PR TITLE
Update validation rules

### PR DESCRIPTION
**Pull request by liweiyi88**

### CODE REVIEW

File: src/Sdk/Validation/Rules/Type.php [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    /**
+     * 'type' rule
+     *
+     * @return void
+     */
+    protected function process(): void
+    {
+        foreach ($this->parameters as $parameter) {
+            if (empty($parameter)) {
+                $this->error = 'type parameter is missing, please specify';
```
Comment (natepage) [inline src/Sdk/Validation/Rules/Type.php]: Please use `\sprintf()` function as much as you can.

**Comment by natepage** [comment]
|`src/Sdk/Validation/Validator.php:136`|
|---|
Please add safety to the usage of indexes, if for any reason `\preg_match_all` doesn't populate `$matches` properly you will get an exception for undefined index that you don't control because doesn't extends your exceptions.

**Comment by natepage** [comment]
|`src/Sdk/Validation/Validator.php:130`|
|---|
This `explode` might introduce a BC to the current implementation of the sdkblueprint.
The issue is with `Regex` validation rule, if you have a rule that defines a regex with a pipe in it this will break down the regex in multiple fragments.
Need to make sure to update the implementations before deploying it.

File: src/Sdk/Validation/Rules/Type.php [hunk 2]
```diff
@@ -0,0 +1,40 @@
+<?php
+declare(strict_types=1);
+
+namespace LoyaltyCorp\SdkBlueprint\Sdk\Validation\Rules;
+
+use LoyaltyCorp\SdkBlueprint\Sdk\Validation\Rule;
```
Comment (natepage) [inline src/Sdk/Validation/Rules/Type.php]: Please add a space after commas in function parameters.

File: src/Sdk/Validation/Rules/Type.php [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+class Type extends Rule
+{
+    /**
```
Comment (natepage) [inline src/Sdk/Validation/Rules/Type.php]: |`line 37`|
|---|
Please use `\sprintf()` everywhere :)

**Comment by natepage** [comment]
Yes completely agree but it means that we have to update the applications using the sdkblueprint to pass an array everytime we have a `regex` validation, not an issue but we have to do it.

**Comment by natepage** [comment]
I think that we are never too safe :) Let's add the safety to avoid having to change the code if one day `\preg_match_all` behaviour changes.

**Comment by liweiyi88** [comment]
I agree that we need to be safe when we use array indexes in our code. However, IMO, the case is different here as the logic 
```
 foreach (\array_keys($matches[0]) as $key) {
                $rules[$matches[1][$key]] = $matches[2][$key];
            }
 ``` is tight too much to the returned value of `preg_match_all`. Thus, if some day, the behavior of `preg_match_all` changed or the regex pattern changed, the code should break and we should updated the code accordingly. If we put the safety check, we may introduce a bug which can be hard to debug as all the tests will pass by the safety check. What do you think?

**Comment by liweiyi88** [comment]
@{d433845b-4824-46f7-a4e5-e198e09c023c,Nathan Page}, I have added one more rule called `in` in order to add validation rule for currency in Pcizone SDK. Please review if you have a chance.

**Pull request merged**